beamer/agent/relayer.py

```diff
@@ -60,9 +60,14 @@ def run_relayer_for_tx(
             ]
         )
 
-    subprocess.run(
+    result = subprocess.run(
         command_args,
         capture_output=True,
         encoding="utf-8",
         check=True,  # check throws an error right away
     )
+
+    if prove_tx:
+        return result.stdout.split()[-1].split(": ")[-1]
```

Comment:
This does not address @GabrielBuragev's concern.

I suggest something like the following (totally untested, please adjust accordingly):
```python
match = re.search('Proof timestamp: (\d+)', result.stdout)
if match is not None:
    return match.group(1)
raise RuntimeError("relayer failed to provide proof timestamp")
```

The reason for the exception being that code in `chain.py` can react to it.